Documentation/contributing/testing/e2e_legacy.rst

```diff
@@ -0,0 +1,739 @@
+constructs:
+
+Test description diagram::
+
+    Describe
+        BeforeAll(A)
+        AfterAll(A)
+        AfterFailed(A)
+        AfterEach(A)
+        JustAfterEach(A)
+        TESTA1
+        TESTA2
+        TESTA3
+        Context
+            BeforeAll(B)
+            AfterAll(B)
+            AfterFailed(B)
+            AfterEach(B)
+            JustAfterEach(B)
+            TESTB1
+            TESTB2
+            TESTB3
+
+
+Test execution flow::
+
+    Describe
+        BeforeAll
+        TESTA1; JustAfterEach(A), AfterFailed(A), AfterEach(A)
+        TESTA2; JustAfterEach(A), AfterFailed(A), AfterEach(A)
+        TESTA3; JustAfterEach(A), AfterFailed(A), AfterEach(A)
+        Context
+            BeforeAll(B)
+            TESTB1:
+               JustAfterEach(B); JustAfterEach(A)
+               AfterFailed(B); AfterFailed(A);
+               AfterEach(B) ; AfterEach(A);
+            TESTB2:
+               JustAfterEach(B); JustAfterEach(A)
+               AfterFailed(B); AfterFailed(A);
+               AfterEach(B) ; AfterEach(A);
+            TESTB3:
+               JustAfterEach(B); JustAfterEach(A)
+               AfterFailed(B); AfterFailed(A);
+               AfterEach(B) ; AfterEach(A);
+            AfterAll(B)
+        AfterAll(A)
+
+Debugging:
+~~~~~~~~~~
+
+You can retrieve all run commands and their output in the report directory
+(``./test/test_results``). Each test creates a new folder, which contains
+a file called log where all information is saved, in case of a failing
+test an exhaustive data will be added.
+
+.. code-block:: shell-session
+
+	$ head test/test_results/RuntimeKafkaKafkaPolicyIngress/logs
+	level=info msg=Starting testName=RuntimeKafka
+	level=info msg="Vagrant: running command \"vagrant ssh-config runtime\""
+	cmd: "sudo cilium status" exitCode: 0
+	 KVStore:            Ok         Consul: 172.17.0.3:8300
+	ContainerRuntime:   Ok
+	Kubernetes:         Disabled
+	Kubernetes APIs:    [""]
+	Cilium:             Ok   OK
+	NodeMonitor:        Disabled
+	Allocated IPv4 addresses:
+
+
+Running with delve
+^^^^^^^^^^^^^^^^^^
+
+`Delve <https://github.com/derekparker/delve>`_ is a debugging tool for Go
+applications. If you want to run your test with delve,  you should add a new
+breakpoint using
+`runtime.BreakPoint() <https://golang.org/pkg/runtime/#Breakpoint>`_ in the
+code, and run ginkgo using ``dlv``.
+
+Example how to run ginkgo using ``dlv``:
+
+.. code-block:: shell-session
+
+	dlv test . -- --ginkgo.focus="Runtime" -ginkgo.v=true --cilium.provision=false
+
+Running End-To-End Tests In Other Environments via kubeconfig
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The end-to-end tests can be run with an arbitrary kubeconfig file. Normally the
+CI will use the kubernetes created via vagrant but this can be overridden with
+``--cilium.kubeconfig``. When used, ginkgo will not start a VM nor compile
+cilium. It will also skip some setup tasks like labeling nodes for testing.
```

Comment:
```suggestion
You can run the end-to-end tests with an arbitrary kubeconfig file. Normally, the
CI uses the Kubernetes instance created with Vagrant, but this can be overridden with
``--cilium.kubeconfig``. When used, Ginkgo does not start a VM nor compile
Cilium. It also skips some setup tasks like labeling nodes for testing.
```